### PR TITLE
Giane/[02] Feat: Masks extension

### PR DESCRIPTION
**Pull request by gianebinotto**
Extension to handle masks for credit card, CPF, CEP, cellphone or any other pattern

### CODE REVIEW

**Review by guilucas07 — APPROVED** [review]

File: core/src/main/java/io/cheesecakelabs/core/utils/extensions/EditText+Maks.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+fun EditText.maskCPF() {
+    mask("###.###.###-##")
+}
+
+fun EditText.maskCEP() {
+    mask("#####-###")
+}
+
+fun EditText.maskCreditCard() {
+    mask("#### #### #### ####")
```
Comment (diegohkd) [inline core/src/main/java/io/cheesecakelabs/core/utils/extensions/EditText+Maks.kt]: What if the card has more than 16 digits?

File: core/src/main/java/io/cheesecakelabs/core/utils/extensions/EditText+Maks.kt [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+fun EditText.maskCellPhone(hasCountryCode: Boolean) {
+    if (hasCountryCode) mask("+## (##) #########")
+    else mask("(##) #########")
```
Comment (diegohkd) [inline core/src/main/java/io/cheesecakelabs/core/utils/extensions/EditText+Maks.kt]: What if the area code has 3 digits?

File: core/src/main/java/io/cheesecakelabs/core/utils/extensions/EditText+Maks.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    this.addTextChangedListener(object : TextWatcher {
+        var isUpdating: Boolean = false
+        var previousString: String = ""
+
+        override fun afterTextChanged(s: Editable?) {}
+
+        override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
+
+        override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+            val str = s.toString().replace("[./() \\-+]".toRegex(), "")
```
Comment (diegohkd) [inline core/src/main/java/io/cheesecakelabs/core/utils/extensions/EditText+Maks.kt]: I don't know if it's a good idea to do something so generic here. Maybe some masks would need one of this characters. One option is to have different TextWatchers for each mask. I remember seeing this on Nobis project: 
https://github.com/CheesecakeLabs/local-android/tree/dev/app/src/main/java/com/nobis/app/utils/text_watchers

**Review by diegohkd — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/cheesecakelabs/core/utils/extensions/EditText+Maks.kt (hunk 1 above)
Comment (gianebinotto) [reply core/src/main/java/io/cheesecakelabs/core/utils/extensions/EditText+Maks.kt]: Couldn't the general mask be used in this case?

**Review by RubensHardt — APPROVED** [review]

**Review by ximendes-regis — APPROVED** [review]

**Pull request merged**